### ISSUE TITLE
Short alias for `git checkout -b`

### ISSUE THREAD

**Issue by icaroseara** [robbyrussell/oh-my-zsh#6156]
`git checkout -b` is a useful idiom.

Yet, there's no short alias for it in the git plugin.

Wouldn't it be useful to add one?

**Comment by BenGitsCode** [comment]
I honestly thought `gcb` was an oh-my-zsh builtin alias? I must have set it myself, but I'm not positive as to where.

**Comment by eremit** [comment]
`gcb` is part of the git plugin.

**Comment by mdrmike** [comment]
**Oh-my-zsh has this as both the default behavior and it is documented.  *To ask someone else for their time* without first using your time to research is an abuse of your power to submit an issue.  The effect is harmful.**

@icaroseara Everyone appreciates good & useful feedback.  In the future, please help yourself before asking for help.  Judging by your Github profile I suspect you may eventually want to be a successful and valued member of an open source project? [If so, Follow these unwritten rules](http://www.infoworld.com/article/3156776/open-source-tools/the-6-unwritten-rules-of-open-source-development.html).

#### You might also check out

-   Please, [Open Source RTFM+RTFC](https://yakovfain.com/2010/02/18/in-open-source-rtfm-means-rtfc/)
-   [How To Be A Good Open Source Community Member](http://spf13.com/post/how-to-be-a-good-open-source-community-member/)

My goal is not to be rude or mean or to waste your time... rather to educate and help you understand when you ask someone for their time, it can be off-putting *if its clear you're not willing to invest that same time to help yourself.*  


## General Rules—When Unsure, RTFM

Most important take-away: Please don't waste someone else's time.  Do your homework first.

Even [opening an issue](https://github.com/robbyrussell/oh-my-zsh/issues/6156) with a community solution still means the developer (or someone on team) needs to read, and close the issue.  This can lead to what's called developer fatigue.  And eventually can lead to someone no longer desiring to share their time/knowledge freely with people who won't be respectful.  

### Documentation 

-   Read about [Oh-my-zsh plugins](https://github.com/robbyrussell/oh-my-zsh#plugins)
-   Read about the [git plugin](https://github.com/robbyrussell/oh-my-zsh/wiki/Plugin:git)




### Solution

Let me share with you how I figured this out.  

1.   I copied the command `git checkout -b` from the [original issue post](6156#1).  
2.   Then I [searched this repository](https://github.com/robbyrussell/oh-my-zsh/search?utf8=%E2%9C%93&q=git+checkout+-b&type=) for that text.  
3.   The first result was for the [git plugin](https://github.com/robbyrussell/oh-my-zsh/blob/456341fd69c3e514e401f1c3c1726b77d733c86b/plugins/git/git.plugin.zsh).

That's all I needed to be sure a plugin already exists... and either you didn't search (which suggests lazy) but giving you the benefit of the doubt just don;t yet know how to use oh-my-zsh.  Go make sure the git plugin is in your list of plugins.

> **Note:** Keep in mind, oh-my-zsh fundamentally changes your shell (read: system).  If you're unsure of what you're doing, test first in a safe environment.  Making system level changes that you're unsure of ramifications or how they work, easily leads to an unstable or unsafe (and often hackable) computer system.


##### TL;DR
Per the [template](https://github.com/robbyrussell/oh-my-zsh/blob/master/templates/zshrc.zsh-template), `plugins=(git)` (on line 54) is the only default plugin when you install oh-my-zsh.  Which also means @BenGitsCode is somewhat correct, in the sense it feels built-in to oh-my-zsh.

**Comment by pinsily** [comment]
hello

At 2017-06-29 07:07:46, "mike stewart" <notifications@github.com> wrote:


Oh-my-zsh has this as both the default behavior and it is documented. To ask someone else for their time without first using your time to research is an abuse of your power to submit an issue. The effect is harmful.

@icaroseara Everyone appreciates good & useful feedback. In the future, please help yourself before asking for help. Judging by your Github profile I suspect you may eventually want to be a successful and valued member of an open source project? If so, Follow these unwritten rules.

You might also check out
Please, Open Source RTFM+RTFC
How To Be A Good Open Source Community Member

My goal is not to be rude or mean or to waste your time... rather to educate and help you understand when you ask someone for their time, it can be off-putting if its clear you're not willing to invest that same time to help yourself.

General Rules—When Unsure, RTFM

Most important take-away: Please don't waste someone else's time. Do your homework first.

Even opening an issue with a community solution still means the developer (or someone on team) needs to read, and close the issue. This can lead to what's called developer fatigue. And eventually can lead to someone no longer desiring to share their time/knowledge freely with people who won't be respectful.

Documentation
Read about Oh-my-zsh plugins
Read about the git plugin
Solution

Let me share with you how I figured this out.

I copied the command git checkout -b from the original issue post.
Then I searched this repository for that text.
The first result was for the git plugin.

That's all I needed to be sure a plugin already exists... and either you didn't search (which suggests lazy) but giving you the benefit of the doubt just don;t yet know how to use oh-my-zsh. Go make sure the git plugin is in your list of plugins.

Note: Keep in mind, oh-my-zsh fundamentally changes your shell (read: system). If you're unsure of what you're doing, test first in a safe environment. Making system level changes that you're unsure of ramifications or how they work, easily leads to an unstable or unsafe (and often hackable) computer system.

TL;DR

Per the template, plugins=(git) (on line 54) is the only default plugin when you install oh-my-zsh. Which also means @BenGitsCode is somewhat correct, in the sense it feels built-in to oh-my-zsh.

**Comment by mcornella** [comment]
Agree on the sentiment, not the tone. It was a harmless mistake.
